### PR TITLE
engine: otel span for cold start

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: internal/engine/cold_start_tracker.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package engine
```
Comment (nicks) [inline internal/engine/cold_start_tracker.go]: all new subscribers should go in a subpackage! might make sense to put this in the same package as the other trace stuff?

File: internal/engine/cold_start_tracker.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+func (c *ColdStartTracker) OnChange(ctx context.Context, st store.RStore) {
+	if c.coldStartFinished {
+		return
+	}
+
+	state := st.RLockState()
+	defer st.RUnlockState()
+
+	if !state.InitialBuildsCompleted() && c.span == nil {
+		_, span := c.tracer.Start(ctx, "first_run")
```
Comment (nicks) [inline internal/engine/cold_start_tracker.go]: fwiw, subscribers have an extra SetUp interface that might be helpful here

**Comment by jazzdan** [comment]
ping @nicks @landism

**Comment by landism** [comment]
It looks like you dropped "cold" from the file names but not the code?

**Comment by jazzdan** [comment]
@landism whoops, thanks goland

File: internal/engine/telemetry/start_tracker.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+	"go.opentelemetry.io/otel/api/trace"
+
+	"github.com/windmilleng/tilt/internal/store"
+)
+
+type StartTracker struct {
+	tracer            trace.Tracer
+	span              trace.Span
+	coldStartFinished bool
```
Comment (landism) [inline internal/engine/telemetry/start_tracker.go]: one more!

**Pull request merged**